### PR TITLE
Add a feature so that participants can't make their submissions private if challenge host desires

### PR DESCRIPTION
**Pull request by Sanji515**
- Added a drop-down for challenge host to toggle the visibility of the submissions of a particular challenge phase in `View All Submissions` tab.

- Used exisiting API `challenge_phase_detail` for updating the challenge phase after toggling the submissions visibility

- After toggling the submissions visibility, `Show on Leaderboard` column from the `My Submissions` tab will be `displayed / not displayed` according the submissions visibility change.

- And also on the leaderboard, entries will be displayed according to the submissions visibility made by challenge host.

**Screenshot:**

Screenshot from 2019-06-16 21-16-10

### CODE REVIEW

**Comment by Sanji515** [comment]
@RishabhJain2018 @galipremsagar Please review

**Comment by galipremsagar** [comment]
Could you add the screenshot of UI aswell @Sanji515 ?

File: frontend/src/js/controllers/challengeCtrl.js [hunk 1]
```diff
@@ -580,67 +582,92 @@
...
-                        if (duration._data.years != 0) {
-                            var years = duration.asYears();
-                            vm.leaderboard[i].submission__submitted_at = years;
-                            if (years.toFixed(0)==1) {
-                                vm.leaderboard[i].timeSpan = 'year';
-                            } else {
-                                vm.leaderboard[i].timeSpan= 'years';
+                    var privateEntriesCount = 0;
+                    $timeout(function () {
+                        for (var i=0; i<vm.leaderboard.length; i++) {
```
Comment (galipremsagar) [inline frontend/src/js/controllers/challengeCtrl.js]: Could you commit indented file changes?

File: frontend/src/views/web/challenge/leaderboard.html [hunk 2]
```diff
@@ -57,7 +57,7 @@
                                     class="result-wrn"><span id="baseline-badge-desc"
                                         class="new badge orange-background w-400" data-badge-caption="B"></span>
                                     <span class="w-400"> - Baseline submission</span></div>
-                                <table ng-if="challenge.isResult && challenge.leaderboard.length"
+                                <table ng-if="challenge.isResult && challenge.leaderboard.length && challenge.allLeaderboardEntriesPrivate"
```
Comment (galipremsagar) [inline frontend/src/views/web/challenge/leaderboard.html]: Why this check? Can you explain what you are trying to achieve here?

File: frontend/src/js/controllers/challengeCtrl.js [hunk 3]
```diff
@@ -580,67 +582,92 @@
...
+                                } else {
+                                    vm.leaderboard[i].timeSpan = 'seconds';
+                                }
                             }
                         }
-                    }
+                        if (vm.leaderboard.length == privateEntriesCount) {
+                            vm.allLeaderboardEntriesPrivate = false;
+                        }
+                    }, 2000);
```
Comment (galipremsagar) [inline frontend/src/js/controllers/challengeCtrl.js]: Why are you wrapping it in a timeout code?

Comment (Sanji515) [reply frontend/src/js/controllers/challengeCtrl.js]: to get the `phaseSplits`, without timeout `phaseSplits` array output as empty

Comment (galipremsagar) [reply frontend/src/js/controllers/challengeCtrl.js]: Why would `phaseSplits` be an empty array? Did you look into the reason?

**Pull request closed without merging**